src/main/java/com/google/codeu/servlets/UserInfoServlet.java

```diff
@@ -0,0 +1,91 @@
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.User;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.BufferedReader;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Handles fetching and saving user data. */
+@WebServlet("/user-info")
+public class UserInfoServlet extends HttpServlet {
+
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /** Responds with the "about me" section for a particular user. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType("application/json");
+
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    if (userEmail == null || userEmail.equals("")) {
+      // Request is invalid, return empty response
```

Comment:
Maybe a good idea is to redirect the user to another view? ...just an option but not necessary.
The idea is to let the user know something went wrong.

Reply:
It is not possible for the user to get to this page without logging in, but I can agree that this is there in case of a failure. So I'll redirect them to the homepage.